src/providers/vehicle-details-by-category/vehicle-details-by-category.ts

```diff
@@ -55,4 +63,51 @@ export class VehicleDetailsByCategoryProvider {
         throw new Error(VehicleDetailsByCategoryProvider.getVehicleDetailsByCategoryCodeErrMsg);
     }
   }
+
+  public getVehicleDetailsOnlyByCategoryCode(category: CategoryCode): any {
```

Comment:
any reason we have not extended the existing getVehicleDetailsByCategoryCode function here? Half of these conditions would already be covered in that function

Reply:
added functionility to existing function